### PR TITLE
Separting the billing address and the mailing address

### PR DESCRIPTION
**Pull request by juggernautsei**
The time has come to add this field to the facilities

### CODE REVIEW

File: interface/usergroup/facilities.php [hunk 1]
```diff
@@ -188,10 +200,24 @@ function refreshme() {
...
+                                        $varmcity=$iter3["mail_city"].",";
+                                    }
+
+                                    if ($iter3["mail_state"]!=""){
+                                        $varmstate=$iter3["mail_state"].",";
+                                    }
                             ?>
                             <tr height="22">
-                                 <td valign="top" class="text"><b><a href="facility_admin.php?fid=<?php echo attr_url($iter3["id"]); ?>" class="medium_modal"><span><?php echo text($iter3["name"]);?></span></a></b>&nbsp;</td>
+                                 <td valign="top" class="text"><b><a href="facility_admin.php?fid=<?php echo attr($iter3["id"]); ?>" class="medium_modal"><span><?php echo text($iter3["name"]);?></span></a></b>&nbsp;</td>
```
Comment (bradymiller) [inline interface/usergroup/facilities.php]: revert the change in line above

File: interface/usergroup/facilities_add.php [hunk 2]
```diff
@@ -287,6 +287,45 @@ function displayAlert()
...
         </tr>
+        <tr>
+            <td>
+                <hr>
+            </td>
+        </tr>
+
+        <tr>
+            <td>
+                <label><?php xl('Mailing Address', 'e'); ?>: </label>
```
Comment (bradymiller) [inline interface/usergroup/facilities_add.php]:
```
<label><?php echo xlt('Mailing Address'); ?>: </label>
```

File: interface/usergroup/facilities_add.php [hunk 3]
```diff
@@ -287,6 +287,45 @@ function displayAlert()
...
+            </td>
+            <td>
+                <input type=entry size=20 name=mail_street value="<?php echo htmlspecialchars($facility["mail_street"], ENT_QUOTES) ?>">
```
Comment (bradymiller) [inline interface/usergroup/facilities_add.php]:
```
<input type=entry size=20 name=mail_street value="<?php echo attr($facility["mail_street"]); ?>">

File: interface/usergroup/facilities_add.php [hunk 4]
```diff
@@ -287,6 +287,45 @@ function displayAlert()
...
+            </td>
+        </tr>
+
+        <tr>
+            <td>
+                <label><?php xl('City', 'e'); ?>: </label>
```
Comment (bradymiller) [inline interface/usergroup/facilities_add.php]:
```
<label><?php echo xlt('City'); ?>: </label>

File: interface/usergroup/facilities_add.php [hunk 5]
```diff
@@ -287,6 +287,45 @@ function displayAlert()
...
+            </td>
+            <td>
+                <input type=entry size=20 name=mail_city value="<?php echo htmlspecialchars($facility["mail_city"], ENT_QUOTES) ?>">
```
Comment (bradymiller) [inline interface/usergroup/facilities_add.php]:
```
<input type=entry size=20 name=mail_city value="<?php echo attr($facility["mail_city"]); ?>">

File: interface/usergroup/facilities_add.php [hunk 6]
```diff
@@ -287,6 +287,45 @@ function displayAlert()
...
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <label><?php xl('State', 'e'); ?>: </label>
```
Comment (bradymiller) [inline interface/usergroup/facilities_add.php]:
```
<label><?php echo xlt('State'); ?>: </label>

File: interface/usergroup/facilities_add.php [hunk 7]
```diff
@@ -287,6 +287,45 @@ function displayAlert()
...
+            </td>
+            <td>
+                <input type=entry size=20 name=mail_state value="<?php echo htmlspecialchars($facility["mail_state"], ENT_QUOTES) ?>">
```
Comment (bradymiller) [inline interface/usergroup/facilities_add.php]:
```
<input type=entry size=20 name=mail_state value="<?php echo attr($facility["mail_state"]); ?>">

File: interface/usergroup/facilities_add.php [hunk 8]
```diff
@@ -287,6 +287,45 @@ function displayAlert()
...
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <label><?php xl('Zip', 'e'); ?>: </label>
```
Comment (bradymiller) [inline interface/usergroup/facilities_add.php]:
```
<label><?php echo xlt('Zip'); ?>: </label>

File: interface/usergroup/facilities_add.php [hunk 9]
```diff
@@ -287,6 +287,45 @@ function displayAlert()
...
+            </td>
+            <td>
+                <input type=entry size=20 name=mail_zip value="<?php echo htmlspecialchars($facility["mail_zip"], ENT_QUOTES) ?>">
```
Comment (bradymiller) [inline interface/usergroup/facilities_add.php]:
```
<input type=entry size=20 name=mail_zip value="<?php echo attr($facility["mail_zip"]); ?>">
```

File: interface/usergroup/facility_admin.php [hunk 10]
```diff
@@ -274,6 +274,45 @@ function displayAlert()
...
+            <td>
+                <input type=entry size=20 name=mail_state value="<?php echo htmlspecialchars($facility["mail_state"], ENT_QUOTES) ?>">
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <label><?php xl('Zip', 'e'); ?>: </label>
+            </td>
+            <td>
+                <input type=entry size=20 name=mail_zip value="<?php echo htmlspecialchars($facility["mail_zip"], ENT_QUOTES) ?>">
```
Comment (bradymiller) [inline interface/usergroup/facility_admin.php]: make all the changes in this script recommended in the above script

File: services/FacilityService.php [hunk 11]
```diff
@@ -324,7 +336,10 @@ private function get($map)
         $sql .= "        FAC.color,";
         $sql .= "        FAC.primary_business_entity,";
         $sql .= "        FAC.facility_code,";
-        $sql .= "        FAC.extra_validation";
```
Comment (bradymiller) [inline services/FacilityService.php]: why was above line removed?

File: sql/5_0_1-to-5_0_2_upgrade.sql [hunk 12]
```diff
@@ -669,3 +669,7 @@ CREATE TABLE `api_token` (
 ) ENGINE = InnoDB;
 #EndIf
 
+#IfMissingColumn users patient_menu_role
```
Comment (bradymiller) [inline sql/5_0_1-to-5_0_2_upgrade.sql]: above line will not work :)

File: sql/5_0_1-to-5_0_2_upgrade.sql [hunk 13]
```diff
@@ -669,3 +669,7 @@ CREATE TABLE `api_token` (
...
+ALTER TABLE `facility` ADD `mail_street` VARCHAR(30) NOT NULL AFTER `facility_taxonomy`, ADD `mail_city` VARCHAR(35) NOT NULL AFTER `mail_street`, ADD `mail_state` VARCHAR(2) NOT NULL AFTER `mail_city`, ADD `mail_zip` VARCHAR(10) NOT NULL AFTER `mail_state`;
```
Comment (bradymiller) [inline sql/5_0_1-to-5_0_2_upgrade.sql]: make newlines between each ADD and do not use AFTER

File: sql/5_0_1-to-5_0_2_upgrade.sql [hunk 14]
```diff
@@ -669,3 +669,7 @@ CREATE TABLE `api_token` (
...
+ALTER TABLE `facility` ADD `oid` INT(15) NOT NULL DEFAULT '0' FIRST;
```
Comment (bradymiller) [inline sql/5_0_1-to-5_0_2_upgrade.sql]: @sjpadgett ,
Can I get clarification on this. saw it discussed in the forum. Does this need to be set in the facility form? Is it only numbers?
thanks, -brady

Comment (bradymiller) [reply sql/5_0_1-to-5_0_2_upgrade.sql]: @juggernautsei ,
Likely no need for FIRST here, but lets wait for Jerry's input.

File: sql/database.sql [hunk 15]
```diff
@@ -1620,33 +1620,37 @@ CREATE TABLE `standardized_tables_track` (
...
-  `attn` varchar(65) default NULL,
-  `domain_identifier` varchar(60) default NULL,
-  `facility_npi` varchar(15) default NULL,
-  `facility_taxonomy` varchar(15) default NULL,
-  `tax_id_type` VARCHAR(31) NOT NULL DEFAULT '',
-  `color` VARCHAR(7) NOT NULL DEFAULT '',
-  `primary_business_entity` INT(10) NOT NULL DEFAULT '0' COMMENT '0-Not Set as business entity 1-Set as business entity',
-  `facility_code` VARCHAR(31) default NULL,
+  `oid` int(15) NOT NULL DEFAULT '0' COMMET 'HIEs CCDA and FHIR an OID is required/wanted',
+  `id` int(11) NOT NULL,
```
Comment (bradymiller) [inline sql/database.sql]: Why was the auto_increment removed?

File: sql/database.sql [hunk 16]
```diff
@@ -1620,33 +1620,37 @@ CREATE TABLE `standardized_tables_track` (
...
-  `x12_sender_id` varchar(25) default NULL,
-  `attn` varchar(65) default NULL,
-  `domain_identifier` varchar(60) default NULL,
-  `facility_npi` varchar(15) default NULL,
-  `facility_taxonomy` varchar(15) default NULL,
-  `tax_id_type` VARCHAR(31) NOT NULL DEFAULT '',
-  `color` VARCHAR(7) NOT NULL DEFAULT '',
-  `primary_business_entity` INT(10) NOT NULL DEFAULT '0' COMMENT '0-Not Set as business entity 1-Set as business entity',
-  `facility_code` VARCHAR(31) default NULL,
+  `oid` int(15) NOT NULL DEFAULT '0' COMMET 'HIEs CCDA and FHIR an OID is required/wanted',
```
Comment (bradymiller) [inline sql/database.sql]: This will break likely since COMMENT is mispelled

**Comment by bradymiller** [comment]
hi @juggernautsei ,
Just did a code review. Will plan to test it on next revision. thanks, -brady

File: interface/usergroup/facilities_add.php (hunk 2 above)
Comment (juggernautsei) [reply interface/usergroup/facilities_add.php]: I was going to remove those but there was other work done where they were not removed. So, I left them.

File: services/FacilityService.php (hunk 11 above)
Comment (juggernautsei) [reply services/FacilityService.php]: Unintentional.

File: sql/5_0_1-to-5_0_2_upgrade.sql (hunk 13 above)
Comment (juggernautsei) [reply sql/5_0_1-to-5_0_2_upgrade.sql]: Ok.

File: sql/5_0_1-to-5_0_2_upgrade.sql (hunk 14 above)
Comment (juggernautsei) [reply sql/5_0_1-to-5_0_2_upgrade.sql]: Yes, he did not clarify and I just left it int

File: sql/database.sql (hunk 15 above)
Comment (juggernautsei) [reply sql/database.sql]: unintentional and it did not come out when I dumped the structure of the table.

File: sql/database.sql (hunk 16 above)
Comment (juggernautsei) [reply sql/database.sql]: yep!

File: sql/5_0_1-to-5_0_2_upgrade.sql (hunk 14 above)
Comment (bradymiller) [reply sql/5_0_1-to-5_0_2_upgrade.sql]: Thanks @sjpadgett ,
@juggernautsei , so sounds like no need for FIRST; ie. it can go at the end. And will need to allow user to enter in this value in the facilities form.

Comment (juggernautsei) [reply sql/5_0_1-to-5_0_2_upgrade.sql]: Ok, I can update the form to add the OID entry. I will post an update tomorrow.

**Comment by bradymiller** [comment]
hi @juggernautsei ,
Were the above issues addressed? Thanks, -brady

File: sql/5_0_1-to-5_0_2_upgrade.sql (hunk 12 above)
Comment (juggernautsei) [reply sql/5_0_1-to-5_0_2_upgrade.sql]: I did not add the api_token table creation lines.

**Comment by juggernautsei** [comment]
Here are the changes that were requested 21e5650

File: interface/usergroup/facilities.php (hunk 1 above)
Comment (bradymiller) [reply interface/usergroup/facilities.php]: need to still address

File: interface/usergroup/facilities_add.php (hunk 2 above)
Comment (bradymiller) [reply interface/usergroup/facilities_add.php]: need to still address

File: interface/usergroup/facilities_add.php (hunk 3 above)
Comment (bradymiller) [reply interface/usergroup/facilities_add.php]: need to still address

File: interface/usergroup/facilities_add.php (hunk 4 above)
Comment (bradymiller) [reply interface/usergroup/facilities_add.php]: need to still address

File: interface/usergroup/facilities_add.php (hunk 5 above)
Comment (bradymiller) [reply interface/usergroup/facilities_add.php]: need to still address

File: interface/usergroup/facilities_add.php (hunk 6 above)
Comment (bradymiller) [reply interface/usergroup/facilities_add.php]: need to still address

File: interface/usergroup/facilities_add.php (hunk 7 above)
Comment (bradymiller) [reply interface/usergroup/facilities_add.php]: need to still address

File: interface/usergroup/facilities.php [hunk 17]
```diff
@@ -52,7 +52,11 @@
       "primary_business_entity" => trim(isset($_POST["primary_business_entity"]) ? $_POST["primary_business_entity"] : ''),
       "facility_npi" => trim(isset($_POST["facility_npi"]) ? $_POST["facility_npi"] : ''),
       "facility_taxonomy" => trim(isset($_POST["facility_taxonomy"]) ? $_POST["facility_taxonomy"] : ''),
-      "facility_code" => trim(isset($_POST["facility_id"]) ? $_POST["facility_id"] : '')
+      "facility_code" => trim(isset($_POST["facility_id"]) ? $_POST["facility_id"] : ''),
+      "mail_street" => trim(isset($_POST[":mail_street"]) ? $_POST["mail_street"] : ''),
+        "mail_city" => trim(isset($_POST["mail_city"]) ? $_POST["mail_city"] : ''),
+        "mail_state" => trim(isset($_POST["mail_state"]) ? $_POST["mail_state"] : ''),
+        "mail_zip" => trim(isset($_POST["mail_zip"]) ? $_POST["mail_zip"] : '')
```
Comment (bradymiller) [inline interface/usergroup/facilities.php]: add line for the oid

File: interface/usergroup/facilities.php [hunk 18]
```diff
@@ -85,7 +89,11 @@
       "primary_business_entity" => trim(isset($_POST["primary_business_entity"]) ? $_POST["primary_business_entity"] : ''),
       "facility_npi" => trim(isset($_POST["facility_npi"]) ? $_POST["facility_npi"] : ''),
       "facility_taxonomy" => trim(isset($_POST["facility_taxonomy"]) ? $_POST["facility_taxonomy"] : ''),
-      "facility_code" => trim(isset($_POST["facility_id"]) ? $_POST["facility_id"] : '')
+      "facility_code" => trim(isset($_POST["facility_id"]) ? $_POST["facility_id"] : ''),
+        "mail_street" => trim(isset($_POST["mail_street"]) ? $_POST["mail_street"] : ''),
+        "mail_city" => trim(isset($_POST["mail_city"]) ? $_POST["mail_city"] : ''),
+        "mail_state" => trim(isset($_POST["mail_state"]) ? $_POST["mail_state"] : ''),
+        "mail_zip" => trim(isset($_POST["mail_zip"]) ? $_POST["mail_zip"] : '')
```
Comment (bradymiller) [inline interface/usergroup/facilities.php]: add line for the oid

File: interface/usergroup/facilities_add.php (hunk 8 above)
Comment (bradymiller) [reply interface/usergroup/facilities_add.php]: need to still address

File: interface/usergroup/facilities_add.php (hunk 9 above)
Comment (bradymiller) [reply interface/usergroup/facilities_add.php]: need to still address

File: interface/usergroup/facilities_add.php [hunk 19]
```diff
@@ -287,6 +287,45 @@ function displayAlert()
...
+            </td>
```
Comment (bradymiller) [inline interface/usergroup/facilities_add.php]: also add a entry here to input the OID

File: interface/usergroup/facility_admin.php (hunk 10 above)
Comment (bradymiller) [reply interface/usergroup/facility_admin.php]: need to still address

**Comment by juggernautsei** [comment]
I have looked at the conflict several times and I can't see it.  When I click the resolve conflicts button, it shows not conflict between the master and this branch unless the conflict is that the entries are not in the master?

**Comment by juggernautsei** [comment]
Ok, I went and peeked at the master and the last lines are not in the master but that is the purpose of this branch to add the lines to the code base. I am confused.

**Comment by bradymiller** [comment]
Continued in following PR:
https://github.com/openemr/openemr/pull/2167
